### PR TITLE
High-level "porcelain" API

### PR DESCRIPTION
**Pull request by charlesdaniels**
This PR tracks work to create a high level interface built on top of the Speakeasy generated SDK that will simplify using EOPA in "happy path" use cases.

## What Is Changing?

* Instead of being rooted at `com.styra.opa.`, the SDK is instead rooted at `com.styra.opa.sdk.`. I think we ultimately want the "porcelain" API at the old root, with the SDK in it's own subpackage beneath.

### CODE REVIEW

File: .genignore [hunk 1]
```diff
@@ -1 +1,3 @@
 LICENSE
+src/main/java/com/styra/opa/Porcelain.java
+src/main/java/com/styra/opa/PorcelainException.java
```
Comment (philipaconrad) [inline .genignore:3]:
```suggestion
src/main/java/com/styra/opa/OPA.java
src/main/java/com/styra/opa/OPAException.java
```
[question]: Should these be updated to the above suggestion? IIRC, Speakeasy will nuke your hard work if the names aren't updated. :sweat_smile:

Comment (charlesdaniels) [reply .genignore:3]: Good catch!

Comment (charlesdaniels) [reply .genignore]: Closing this as I did this a while ago

File: README.md [hunk 2]
```diff
@@ -1,7 +1,170 @@
...
+
+Try changing the policy input in `App.java`, or modifying the `main.rego` to observe how the behavior will change. Remember that you will need to restart `opa` any time you edit `main.rego`.
+
+## License
+
 [![License](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
+
+
+*content below this lines is managed by Speakeasy's automated generation tools*
```
Comment (srenatus) [inline README.md:164]: Hmm. I'd prefer if we found a better way. I don't really want us to disclaim ownership of the docs like that. If the generated docs are ~trash~ insufficient, that's our problem to deal with, we can't say that it's SE's generated docs and be done with it.

We can remote sections:

> If you would like to take over the management of automatically generated sections, then you can do the following:
> 
>   1.  Remove the  comment and change it to , which marks that section as managed by you. (This step is important. If you only remove the "Start" comment, the section may be re-inserted as described below.)
>   3.  Edit the content that was between those comments as you see fit.
>

Comment (charlesdaniels) [reply README.md:164]: Yeah that’s a fair point. TBH I should probably just integrate some usage example of the SE generated API and any other pertinent info in the generated readme, then remove the generated sections. 

Should probably make some tests that will trip if the API changes too.

Comment (charlesdaniels) [reply README.md]: I went ahead and adopted the exact same way you had it.

**Pull request merged**